ddtrace/tracer/logger.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import "github.com/DataDog/dd-trace-go/v2/internal/log"
+
+// Logger implementations are able to log given messages that the tracer or profiler might output.
+type Logger interface {
+	// Log prints the given message.
+	Log(msg string)
+}
+
+// UseLogger sets l as the logger for all tracer and profiler logs.
+func UseLogger(l Logger) {
+	log.UseLogger(l)
+}
+
+// LogLevel represents the logging level that the log package prints at.
+type LogLevel log.Level
```

Comment:
💅 NIT: Maybe use a type alias here? Makes the code below a little simpler:

<details>
<summary>Patch</summary>

```diff
diff --git i/ddtrace/tracer/logger.go w/ddtrace/tracer/logger.go
index 43d2e18c5..c57573410 100644
--- i/ddtrace/tracer/logger.go
+++ w/ddtrace/tracer/logger.go
@@ -19,18 +19,14 @@ func UseLogger(l Logger) {
 }
 
 // LogLevel represents the logging level that the log package prints at.
-type LogLevel log.Level
-
-func (l LogLevel) String() string {
-	return log.Level(l).String()
-}
+type LogLevel = log.Level
 
 type loggerAdapter struct {
 	fn func(lvl LogLevel, msg string, a ...any)
 }
 
 func (l loggerAdapter) Log(msg string) {
-	l.fn(LogLevel(log.CurrentLevel()), msg)
+	l.fn(log.CurrentLevel(), msg)
 }
 
 // AdaptLogger adapts a function to the Logger interface to adapt any logger

</details>

```

Reply:
Nice trick, I wasn't aware of it! Thanks @felixge!